fvtest/tril/tril/CMakeLists.txt

```diff
@@ -23,44 +23,88 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake/modules"
 	${CMAKE_MODULE_PATH})
 
 include(OmrUtility)
-include(FindYACC)
-include(FindLEX)
-
-find_package(YACC REQUIRED)
-find_package(LEX REQUIRED)
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-list(APPEND TRIL_FLEX_FLAGS)
+find_package(YACC)
+find_package(LEX)
+
+if (YACC AND LEX)
+
+	include(FindYACC)
+	include(FindLEX)
+
+	list(APPEND TRIL_FLEX_FLAGS)
+
+	# There is no unistd.h on Windows, so if win_flex.exe is in use,
+	# the --wincompat option must be added.
+	if(OMR_HOST_OS STREQUAL "win")
+		list(APPEND TRIL_FLEX_FLAGS
+			"--wincompat"
+		)
+	endif()
+
+	omr_stringify(TRIL_FLEX_OPTIONS ${TRIL_FLEX_FLAGS})
+	YACC_TARGET(tril_parser tril.y ${CMAKE_CURRENT_BINARY_DIR}/tril.parser.c
+		DEFINES_FILE ${CMAKE_CURRENT_BINARY_DIR}/tril.parser.h)
+	LEX_TARGET(tril_scanner tril.l ${CMAKE_CURRENT_BINARY_DIR}/tril.scanner.c
+		COMPILE_FLAGS ${TRIL_FLEX_OPTIONS}
+	)
+	ADD_LEX_YACC_DEPENDENCY(tril_scanner tril_parser)
+
+	list(APPEND TRIL_SOURCES 
+		${YACC_tril_parser_OUTPUTS}
+		${LEX_tril_scanner_OUTPUTS}
+	)
+
+#If LEX and YACC is not found, then look for Bison and Flex
+else() # LEX AND YACC
+
+	find_package(BISON REQUIRED)
+	find_package(FLEX REQUIRED)
 
-# There is no unistd.h on Windows, so if win_flex.exe is in use,
-# the --wincompat option must be added.
-if(OMR_HOST_OS STREQUAL "win")
 	list(APPEND TRIL_FLEX_FLAGS
-		"--wincompat"
+		--yylineno
+		--header-file=${CMAKE_CURRENT_BINARY_DIR}/tril.scanner.h
 	)
+
+	# There is no unistd.h on Windows, so if win_flex.exe is in use,
+	# the --wincompat option must be added.
+	if(OMR_HOST_OS STREQUAL "win")
+		list(APPEND TRIL_FLEX_FLAGS
+			"--wincompat"
+		)
 endif()
 
 omr_stringify(TRIL_FLEX_OPTIONS ${TRIL_FLEX_FLAGS})
-YACC_TARGET(tril_parser tril.y ${CMAKE_CURRENT_BINARY_DIR}/tril.parser.c
-	DEFINES_FILE ${CMAKE_CURRENT_BINARY_DIR}/tril.parser.h)
-LEX_TARGET(tril_scanner tril.l ${CMAKE_CURRENT_BINARY_DIR}/tril.scanner.c
-	COMPILE_FLAGS ${TRIL_FLEX_OPTIONS}
+BISON_TARGET(tril_parser tril.y ${CMAKE_CURRENT_BINARY_DIR}/tril.parser.c)
+FLEX_TARGET(tril_scanner tril.l ${CMAKE_CURRENT_BINARY_DIR}/tril.scanner.c
+    COMPILE_FLAGS ${TRIL_FLEX_OPTIONS}
```

Comment:
I know that the command invocations above originally used uppercase function names, but since you're making a change here, I'd rather any new code stick to the convention of lowercase names for command names. (command names are case-insensitive in cmake).

Reply:
Are you talking about `FLEX_TARGET` for example? CMake itself seems to use uppercase [1].

[1] https://github.com/Kitware/CMake/blob/master/Modules/FindFLEX.cmake

Reply:
Yep! Older cmake documentation tends to use uppercase command names, but we generally use lowercase (eg `add_library(...)`).